### PR TITLE
Use new SDK based on crosstool-ng

### PR DESCRIPTION
**Pull request by nashif**
Changes to allow using SDK available at

https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.10-beta3

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

File: cmake/toolchain/zephyr/generic.cmake [hunk 1]
```diff
@@ -1,7 +1,9 @@
+set(TOOLCHAIN_HOME ${ZEPHYR_SDK_INSTALL_DIR})
+
 set(COMPILER gcc)
 
 set(CROSS_COMPILE_TARGET i586-${TOOLCHAIN_VENDOR}-elf)
 set(SYSROOT_TARGET       i586-${TOOLCHAIN_VENDOR}-elf)
 
-set(CROSS_COMPILE ${TOOLCHAIN_HOME}/usr/bin/${CROSS_COMPILE_TARGET}/${CROSS_COMPILE_TARGET}-)
+set(CROSS_COMPILE ${TOOLCHAIN_HOME}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
```
[the remark this answers is not in the document]
Comment (nashif) [reply cmake/toolchain/zephyr/generic.cmake]: how is all of that related to x86_64? That change might not be ready yet and needs to address a few more cases.
And yes, there is a strong motivation to change the structure, this is a completely different toolchain built with a different tool and what we had before was very wrong and weird (the directory structure) and we never liked it, but that is what we got from yocto.

[the remark this answers is not in the document]
Comment (pabigot) [reply cmake/toolchain/zephyr/generic.cmake]: On the surface it looks like the path change would require you to change SDKs if you were bisecting across the commit where this gets merged.  That'd be annoying.

[the remark this answers is not in the document]
Comment (nashif) [reply cmake/toolchain/zephyr/generic.cmake]: > On the surface it looks like the path change would require you to change SDKs if you were bisecting across the commit where this gets merged. That'd be annoying.

yes, this is a problem we agreed we want to provide a solution for. Right now this is just the first commit of many to come.

[the remark this answers is not in the document]
Comment (galak) [reply cmake/toolchain/zephyr/generic.cmake]: Yeah, we really should do something based on ${SDK_VERSION} so we can support both 0.10 and 0.9

**Comment by nashif** [comment]
This now requires https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.10.0-beta5 (still building)

File: cmake/toolchain/zephyr/generic.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (nashif) [reply cmake/toolchain/zephyr/generic.cmake]: > Yeah, we really should do something based on ${SDK_VERSION} so we can support both 0.10 and 0.9

this is now done

**Review by carlescufi — APPROVED** [review]

**Review by galak — APPROVED** [review]

File: cmake/toolchain/zephyr/host-tools.cmake [hunk 2]
```diff
@@ -26,7 +26,9 @@ if(NOT (EXISTS ${sdk_version_path}))
 Is ZEPHYR_SDK_INSTALL_DIR=${ZEPHYR_SDK_INSTALL_DIR} misconfigured?")
 endif()
 
-file(READ ${sdk_version_path} SDK_VERSION)
+file(READ ${sdk_version_path} SDK_VERSION_PRE1)
```
Comment (galak) [inline cmake/toolchain/zephyr/host-tools.cmake]: Can you add a comment here about how you are manipulating the SDK_VERSION

**Review by galak — CHANGES_REQUESTED** [review]
Generally looks good. 2 items:

1. Add a comment about the SDK_VERSION manipulation
2. What happens when we have 0.10.1 or 0.11.0 of the SDK (that is still crosstool-ng based)

**Comment by nashif** [comment]
> 2\. What happens when we have 0.10.1 or 0.11.0 of the SDK (that is still crosstool-ng based)

at some point we will raise the min. required version, after everyone has transitioned. This is for moving everyone to the next SDK. 

If we want to keep this model and support dot releases probably We can have the versioned directory be 0.10.x or something like that.

**Comment by galak** [comment]
> at some point we will raise the min. required version, after everyone has transitioned. This is for moving everyone to the next SDK.
> 
> If we want to keep this model and support dot releases probably We can have the versioned directory be 0.10.x or something like that.

I'm fine w/merging this to make progress w/the new SDK, just wanted to raise the issue/comment right now.

If you can add the comment about SDK_VERSION manipulation I'll approve.

**Comment by nashif** [comment]
> If you can add the comment about SDK_VERSION manipulation I'll approve.

done

**Review by galak — APPROVED** [review]

**Pull request merged**